File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/ConnectionAbortedException.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/** The exception that is thrown when an operation fails because the connection has been aborted. */
public final class ConnectionAbortedException extends LocalException {
    /**
     * Constructs a ConnectionAbortedException.
     *
     * @param message the detail message
     * @param closedByApplication {@code true} if the connection was aborted by the application,
     *     {@code false} if the connection was aborted by the Ice runtime.
     */
    public ConnectionAbortedException(String message, boolean closedByApplication) {
        super(message);
        this.closedByApplication = closedByApplication;
    }

    @Override
    public String ice_id() {
        return "::Ice::ConnectionAbortedException";
    }

    /** Indicates whether the connection was closed by the application. */
    public final boolean closedByApplication;

    private static final long serialVersionUID = 7912128468475443570L;
}

File: matlab/lib/+Ice/ConnectionAbortedException.m
classdef ConnectionAbortedException < Ice.LocalException
    %CONNECTIONABORTEDEXCEPTION The exception that is thrown when an operation fails because the connection has been
    %   aborted.
    %
    %   ConnectionAbortedException Properties:
    %     closedByApplication - True if the connection was closed by the application, false if it was closed by the Ice runtime.

    % Copyright (c) ZeroC, Inc.

    properties
        %CLOSEDBYAPPLICATION True if the connection was closed by the application, false if it was closed by the Ice
        %   runtime.
        %   logical scalar
        closedByApplication (1, 1) logical = false
    end
    methods (Hidden)
        function obj = ConnectionAbortedException(closedByApplication, errID, what)
            if nargin == 0 % default constructor
                superArgs = {};
            else
                assert(nargin == 3, 'Invalid number of arguments');
                superArgs = {errID, what};
            end
            obj@Ice.LocalException(superArgs{:});
            if nargin > 0
                obj.closedByApplication = closedByApplication;
            end
        end
    end
end
